src/App.tsx

```diff
@@ -15,37 +18,127 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+// export const App: React.FC = () => {
+//   return (
+//     <div className="section content">
+//       <div className="buttons">
+//         <button type="button" className="button is-info is-light">
+//           Sort alphabetically
+//         </button>
+
+//         <button type="button" className="button is-success is-light">
+//           Sort by length
+//         </button>
+
+//         <button type="button" className="button is-warning is-light">
+//           Reverse
+//         </button>
+
+//         <button type="button" className="button is-danger is-light">
+//           Reset
+//         </button>
+//       </div>
+
+//       <ul>
+//         <ul>
+//           <li data-cy="Good">Dumplings</li>
+//           <li data-cy="Good">Carrot</li>
+//           <li data-cy="Good">Eggs</li>
+//           <li data-cy="Good">Ice cream</li>
+//           <li data-cy="Good">Apple</li>
+//           <li data-cy="Good">...</li>
+//         </ul>
+//       </ul>
+//     </div>
+//   );
+// };
+
+const SORT_FIELD_ALPHABET = 'alph';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  isReversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1[sortField] - good2[sortField];
```

Comment:
The sorting logic for `SORT_FIELD_LENGTH` is incorrect. You should compare the lengths of the strings using `good1.length - good2.length` instead of `good1[sortField] - good2[sortField]`.